osu.Game/Users/UserPanel.cs

```diff
@@ -217,33 +98,105 @@ protected override void LoadComplete()
+
+        protected SpriteIcon CreateStatusIcon() => statusIcon = new SpriteIcon
+        {
+            Icon = FontAwesome.Regular.Circle,
+            Size = new Vector2(25)
+        };
+
+        protected FillFlowContainer CreateStatusMessage(bool rightAlignedChildren)
+        {
+            var statusContainer = new FillFlowContainer
             {
-                statusBar.ResizeHeightTo(0f, transition_duration, Easing.OutQuint);
-                statusBar.FadeOut(transition_duration, Easing.OutQuint);
-                this.ResizeHeightTo(height - status_height, transition_duration, Easing.OutQuint);
-            }
-            else
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Vertical
+            };
+
+            var alignment = rightAlignedChildren ? Anchor.x2 : Anchor.x0;
+
+            statusContainer.Add(lastVisitMessage = new TextFlowContainer(t => t.Font = OsuFont.GetFont(size: 15)).With(text =>
             {
-                statusBar.ResizeHeightTo(status_height, transition_duration, Easing.OutQuint);
-                statusBar.FadeIn(transition_duration, Easing.OutQuint);
-                this.ResizeHeightTo(height, transition_duration, Easing.OutQuint);
-            }
+                text.Anchor = Anchor.y1 | alignment;
+                text.Origin = Anchor.y1 | alignment;
+                text.AutoSizeAxes = Axes.Both;
+                text.Alpha = 0;
+
+                if (User.LastVisit.HasValue)
+                {
+                    text.AddText(@"Last seen ");
+                    text.AddText(new DrawableDate(User.LastVisit.Value, italic: false));
+                }
+            }));
+
+            statusContainer.Add(statusMessage = new OsuSpriteText
+            {
+                Anchor = Anchor.y1 | alignment,
+                Origin = Anchor.y1 | alignment,
+                Font = OsuFont.GetFont(size: 17)
+            });
+
+            return statusContainer;
+        }
 
-            if (status is UserStatusOnline && activity != null)
+        private void displayStatus(UserStatus status, UserActivity activity = null)
+        {
+            if (status != null)
             {
-                statusMessage.Text = activity.Status;
-                statusBg.FadeColour(activity.GetAppropriateColour(colours), 500, Easing.OutQuint);
+                if (activity != null && !(status is UserStatusOffline))
+                {
+                    statusMessage.Text = activity.Status;
+                    statusIcon.FadeColour(activity.GetAppropriateColour(colours), 500, Easing.OutQuint);
+                }
+                else
+                {
+                    lastVisitMessage.FadeTo(status is UserStatusOffline && User.LastVisit.HasValue ? 1 : 0);
+                    statusMessage.Text = status.Message;
+                    statusIcon.FadeColour(status.GetAppropriateColour(colours), 500, Easing.OutQuint);
+                }
+
+                return;
             }
```

Comment:
This needs to be split into a method, refactored for readability (reduce nesting), and also thoroughly tested in the test scene.